Sparkle/SUTouchBarButtonGroup.m

```diff
@@ -0,0 +1,60 @@
+//
+//  SUTouchBarButtonGroup.m
+//  Sparkle
+//
+//  Created by Yuxin Wang on 05/01/2017.
+//  Copyright © 2017 Sparkle Project. All rights reserved.
+//
+
+#import "SUTouchBarButtonGroup.h"
+
+@implementation SUTouchBarButtonGroup
+
+@synthesize buttons = _buttons;
+
+- (instancetype)initByReferencingButtons:(NSArray<NSButton *> *)buttons
```

Comment:
We shouldn't imitate NSAlert exactly (for that reason alone). NSAlert is also not a view controller. I'd stick with the single initializer and not go for more mutability (which should be avoided when possible). This is also a private API for us, not one that we are exposing, so we don't have to worry about certain future implications. This is also why we don't need to expose properties for alternative and other buttons, or support more than three buttons.